.github/templates/free-space/action.yaml

```diff
@@ -0,0 +1,23 @@
+name: free-space
+description: |
+  This action frees up disk space on the runner.
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Remove Android library
+      shell: bash
+      run: sudo rm -rf /usr/local/lib/android
+
+    - name: Remove .NET runtime
+      shell: bash
+      run: sudo rm -rf /usr/share/dotnet
+
+    - name: Remove CodeQL
+      shell: bash
+      run: sudo rm -rf /opt/hostedtoolcache/CodeQL
+
+    - name: Removes cached Docker images
```

Comment:
super minor, i think we just need this one.

Reply:
Just this one is not enough. The previous free-space action from marketplace we used allowed us to free 14gb by removing  android and dotnet. Unfortunately, that was not enough. We had just enough space to run tests, but not to [collect cache](https://github.com/paritytech/trappist/actions/runs/5291713912/jobs/9577645273#step:15:7). (sadly github didn't notify us about it). So we need extra 6gb to pack the artifacts. Docker images and codeql give us 5gb each.

These were the easiest and fastest things to remove. Currently, we have 4gb of free space just in case. If we need more it will harder to get, but still possible.

Reply:
@evilrobot-01 Yes, free github runners run on Azure's [Standard_DS2_v2](https://learn.microsoft.com/en-us/azure/virtual-machines/dv2-dsv2-series#dsv2-series) which have very limited resources. If we talk about disk space, we have 84gb root disk and 14gb temp disk mounted on `/mnt` (for swap). The root disk contains preinstalled software leaving only ~26gb free, which is not enough to build and test trappist. So, we have to remove some of preinstalled software that we aren't going to use.